### PR TITLE
closable chat and error on disconnect

### LINKED ISSUE #174: Show "Connection to chat lost" indicator
When the WebSocket connection dies for some reason there should be a small indicator like `connection to chat lost, reconnecting...`

### LINKED ISSUE #175: Allow hiding chat
As the chat tends to be a little off-topic from time to time, students who want to focus should be able to hide it.

### PR DESCRIPTION
**Pull request by MatthiasReumann**
- Closes Issues #174 and #175 
- With the tests I made the chat window didn't show on mobile/smaller width devices, fixed that
- Chat State (Open/Close) stored in LocalStorage
- UI improvements

### CODE REVIEW

**Comment by joschahenningsen** [comment]
This looks great! 
can the send button be next to the chat input when not on mobile? 
image

File: web/ts/watch.ts [hunk 1]
```diff
@@ -26,6 +26,10 @@ function startWebsocket() {
     if (cf !== null && cf != undefined) {
         (document.getElementById("chatForm") as HTMLFormElement).addEventListener("submit", (e) => submitChat(e));
     }
+    ws.onopen = function (e) {
+        hideDisconnectMsg();
+    };
```
Comment (joschahenningsen) [inline web/ts/watch.ts]: I'd initialize this hidden so the red message doesn't flicker up every time a page is opened.

File: web/ts/watch.ts [hunk 2]
```diff
@@ -129,4 +138,12 @@ function submitChat(e: Event) {
     return false; //prevent form submission
 }
 
+function showDisconnectMsg() {
+    document.getElementById("disconnectMsg").style.display = "block";
+}
+
+function hideDisconnectMsg() {
+    document.getElementById("disconnectMsg").style.display = "none";
+}
```
Comment (joschahenningsen) [inline web/ts/watch.ts]: Nit: You can add and remove the `hidden` & `block` classes here

File: web/template/partial/stream/chat.gohtml [hunk 3]
```diff
@@ -1,14 +1,36 @@
 {{define "chat"}}
     {{- /*gotype: TUM-Live/web.WatchPageData*/ -}}
     {{$stream := .IndexData.TUMLiveContext.Stream}}
-    <div id="chatWrapper"
-         class="hidden md:flex flex-col w-full md:h-full border-t-2 mt-2 md:mt-0 md:border-t-0 md:dark:border-l dark:border-secondary dark:bg-secondary-lighter bg-gray-100 md:w-2/6 lg:w-1/6 text-1 pt-3">
-        <div id="chatBox" class="overflow-scroll overflow-x-hidden overflow-y-scroll" style="flex: 2;">
+    <script src="/static/assets/ts-dist/chat.js?v={{.IndexData.VersionTag}}"></script>
+    <div x-data="{show: false}" x-init="show = initChat();" id="chatWrapper"
```
Comment (joschahenningsen) [inline web/template/partial/stream/chat.gohtml]: This should help with the flickering when loading the website (alpine hides the chat until it's done initializing the page: https://alpinejs.dev/directives/cloak)

```suggestion
    <div x-cloak x-data="{show: false}" x-init="show = initChat();" id="chatWrapper"
```

**Review by joschahenningsen — CHANGES_REQUESTED** [review]
Overall I'm super happy with this, great job :)

Just one more thing I noticed: when loading the page, the last message is not visible because it's hidden behind the input box. This probably happens because the input becomes visible (through alpine) after the chat was scrolled down by js. Maybe the scrolling should be taken care of in alpine? (I'm not seeing this issue every time)

Could you also resolve the conflicts with `dev` (see #217)

File: web/template/partial/stream/chat.gohtml (hunk 3 above)
Comment (MatthiasReumann) [reply web/template/partial/stream/chat.gohtml]: Thanks! Works like a charm.

**Review by joschahenningsen — APPROVED** [review]
This looks super nice :)

**Pull request merged**